src/emu-i386/simx86/syncpu.h

```diff
@@ -103,10 +103,11 @@ typedef struct {
 /*60*/	unsigned short sigalrm_pending, sigprof_pending;
 /*64*/	unsigned int StackMask;
 /*68*/ 	unsigned int mem_base;
+/*6c*/ 	unsigned int df_increments; /* either 0x040201 or 0xfcfeff */
 	/* begin of cr array */
-/*6c*/	unsigned int cr[5]; /* only cr[0] is used in compiled code */
+/*70*/	unsigned int cr[5]; /* only cr[0] is used in compiled code */
 /* ------------------------------------------------ */
-/*80*/	unsigned int end_mark[0];
+/*80*/	//unsigned int end_mark[0] = cr[4]
```

Comment:
I just fetched upstream again after a pause, and saw the issues you had with these offsets. We can't use word offsets in the JIT, it's either byte or dword. Changing to dword increases code size significantly.
Perhaps a compromise is to get rid of the negative offsets and only use byte offsets for the most-often used fields (e.g. CPU regs, segment bases), dword for others.